### PR TITLE
SideNavigation: implement Sandpack live code examples

### PR DESCRIPTION
**Pull request by AlbertCarreras**
#### What changed?

SideNavigation: implement Sandpack live code examples

### CODE REVIEW

File: docs/docs-components/Header.js [hunk 1]
```diff
@@ -23,10 +23,10 @@ import trackButtonClick from './buttons/trackButtonClick.js';
 import { useNavigationContext } from './navigationContext.js';
 import { convertNamesForURL } from './DocsSideNavigation.js';
 
-const PAGE_HEADER_ZINDEX = new FixedZIndex(10);
-const ABOVE_PAGE_HEADER_ZINDEX = new CompositeZIndex([PAGE_HEADER_ZINDEX]);
-
 function Header() {
+  const PAGE_HEADER_ZINDEX = new FixedZIndex(10);
```
Comment (dangerismycat) [inline docs/docs-components/Header.js:27]: These don't need to be defined within the component — any reason for moving them?

File: docs/docs-components/SandpackExample.js [hunk 2]
```diff
@@ -26,18 +26,26 @@ async function copyCode({ code }: {| code: ?string |}) {
...
   name,
   layout,
-  previewHeight,
+  previewHeight = 'md',
   showEditor,
 }: {|
   name: string,
   layout: 'row' | 'column',
-  previewHeight?: number,
+  previewHeight?: 'sm' | 'md' | number | string,
```
Comment (dangerismycat) [inline docs/docs-components/SandpackExample.js:34]: Can this be `'sm' | 'md' | number`?

File: docs/docs-components/SandpackExample.js [hunk 3]
```diff
@@ -94,7 +102,7 @@ export default function SandpackExample({
   code: ?string | (() => Node),
   layout?: 'row' | 'column',
   name: string,
-  previewHeight?: number,
+  previewHeight?: 'sm' | 'md' | number | string,
```
Comment (dangerismycat) [inline docs/docs-components/SandpackExample.js:105]: ditto here

File: docs/pages/components/web/sidenavigation.js [hunk 4]
```diff
@@ -514,301 +301,133 @@ function Example() {
...
-`}
+            }
           />
         </MainSection.Subsection>
         <MainSection.Subsection
           title="Footer"
           description="Footers allow for filters, additional external links or other UI controls to be included at the bottom of the navigation."
         >
           <MainSection.Card
+            // sandpackExample={
```
Comment (dangerismycat) [inline docs/pages/components/web/sidenavigation.js:336]: Can we remove this commented-out code?

File: docs/examples/sidenavigation/activeItemExample.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+import { SideNavigation } from 'gestalt';
+
+export default function Example(): Node {
+  const reactRouterPath = '/sidenavigation';
+
+  return (
+    <SideNavigation accessibilityLabel="Active item example">
+      <SideNavigation.Section label="Navigation">
+        <SideNavigation.TopItem
+          href="https://gestalt.pinterest.systems/pageheader"
```
Comment (dangerismycat) [inline docs/examples/sidenavigation/activeItemExample.js:12]: I don't think we want to use real URLs for the examples, as it causes the user to navigate to the docs site within the Sandpack iframe, which is very confusing. Let's just use `href="#"` for all examples with links.

File: docs/pages/components/web/sidenavigation.js (hunk 4 above)
Comment (AlbertCarreras) [reply docs/pages/components/web/sidenavigation.js:336]: It's a reminder that we don't support gestalt-datepicker in Sandpack. As soon as we do so, we can use this example.

File: docs/examples/sidenavigation/activeItemExample.js (hunk 5 above)
Comment (AlbertCarreras) [reply docs/examples/sidenavigation/activeItemExample.js:12]: yeah, makes sense

File: docs/docs-components/SandpackExample.js (hunk 2 above)
Comment (AlbertCarreras) [reply docs/docs-components/SandpackExample.js:34]: DONE

File: docs/examples/sidenavigation/activeItemExample.js (hunk 5 above)
Comment (AlbertCarreras) [reply docs/examples/sidenavigation/activeItemExample.js:12]: DONE

**Comment by ayeshakmaz** [comment]
Can we prevent the code from showing on the Don't examples?

File: docs/docs-components/SandpackExample.js [hunk 6]
```diff
@@ -94,7 +102,7 @@ export default function SandpackExample({
   code: ?string | (() => Node),
   layout?: 'row' | 'column',
   name: string,
-  previewHeight?: number,
+  previewHeight?: 'sm' | 'md' | number,
   showEditor?: boolean,
```
Comment (ayeshakmaz) [inline docs/docs-components/SandpackExample.js:106]: I think we need an option to hide all the buttons for the "Don't" examples

Comment (AlbertCarreras) [reply docs/docs-components/SandpackExample.js:114]: DONE

**Review by ayeshakmaz — APPROVED** [review]

**Pull request merged**